app/controllers/TraderAddressCorrectController.scala

```diff
@@ -55,15 +53,12 @@ class TraderAddressCorrectController @Inject()(identify: IdentifierAction,
     val form = request.userAnswers.get(TraderAddressCorrectPage).fold(formProvider()) {
       formProvider().fill
     }
-    eoriDetailsService.retrieveEoriDetails(request.eori).flatMap {
-      case Right(eoriDetails) =>
-        for {
-          updatedAnswers <- Future.fromTry(request.userAnswers.set(KnownEoriDetailsPage, eoriDetails))
-          _ <- sessionRepository.set(updatedAnswers)
-        } yield Ok(view(form, eoriDetails.address, backLink()))
-      case Left(error) =>
-        logger.error(error.message + " " + error.status)
-        Future.successful(NotFound(error.message + " " + error.status))
+    request.userAnswers.get(KnownEoriDetailsPage) match {
+      case Some(eoriDetails) =>
+        Future.successful(Ok(view(form, eoriDetails.address, backLink())))
+      case None =>
+        logger.warn("Requested the trader address page without EORI details")
```

Comment:
Probably wants to be an error level as we cannot do anything now except throw a Technical Difficulties page.